modules/cas_cache/threads.c

```diff
@@ -89,11 +89,27 @@ static int _cas_cleaner_thread(void *data)
 	ocf_cleaner_set_cmpl(c, _cas_cleaner_complete);
 
 	do {
+		if (atomic_read(&info->stop))
+			break;
+
+		atomic_set(&info->kicked, 0);
 		init_completion(&info->sync_compl);
 		ocf_cleaner_run(c, cache_priv->io_queues[smp_processor_id()]);
 		wait_for_completion(&info->sync_compl);
-	} while (0 == wait_event_interruptible_timeout(info->wq,
-			atomic_read(&info->stop), msecs_to_jiffies(ms)));
+
+		/*
+		 * In case of nop cleaning policy we don't want to perform cleaning
+		 * until cleaner_kick() is called.
+		 */
+		if (OCF_CLEANER_DISABLE == ms) {
```

Comment:
Please don't use reversed comparison. Modern age compilers are smart enough to warn about accidental assignment in `if` statement.